MongoDB.Entities/Relationships/JoinRecord.cs

```diff
@@ -1,21 +1,30 @@
-﻿namespace MongoDB.Entities
+﻿using MongoDB.Bson.Serialization.Attributes;
+using System;
+
+namespace MongoDB.Entities;
+
+/// <summary>
+/// Represents a parent-child relationship between two entities.
+/// <para>TIP: The ParentID and ChildID switches around for many-to-many relationships depending on the side of the relationship you're accessing.</para>
+/// </summary>
+public class JoinRecord<TId1, TId2> : Entity<(TId1 ParentID, TId2 ChildID)>
```

Comment:
does this mean existing data (ParentID/ChildID values) would have to be migrated to a composite key?
if so, we're gonna have to provide ppl with the migration classes as well 🤔

Reply:
that, or write a custom serializer for that specific class to maintain compatibility

Reply:
yeah serializer sounds good.
dang, really wish i could help you out with this refactor but i'm bloody swamped these days.
hopefully in a week or so i should be able to spend some time on this.
but so far it seems like you've got this ;-)